src/multibody/costs/centroidal-momentum.cpp

```diff
@@ -61,20 +62,20 @@ void CostModelCentroidalMomentum::calcDiff(const boost::shared_ptr<CostDataAbstr
   Eigen::Ref<pinocchio::Data::Matrix6x> Rv = data->Rx.rightCols(nv);
 
   activation_->calcDiff(data->activation, data->r, recalc);
-  pinocchio::getCentroidalDynamicsDerivatives(state_->get_pinocchio(), *data->pinocchio, Rq, d->dhd_dq, d->dhd_dv, Rv);
+  pinocchio::getCentroidalDynamicsDerivatives(state_->get_pinocchio(), *d->pinocchio, Rq, d->dhd_dq, d->dhd_dv, Rv);
 
   // The derivative computation in pinocchio does not take the frame of reference into
```

Comment:
Which quantity do you want to compute here?

Reply:
It is because h_ref is expressed in the frame aligned with the world coordinates system expressed around the world frame?

Reply:
If my statement is valid, then:
```
data->Rx.block<3, 1>(3, i) -= d->pinocchio->Jcom.col(i).cross(d->pinocchio->hg.linear()); 
```
and should be replaced by:
```
     data->Rx.block<3, 1>(3, i) -= d->pinocchio->Jcom.col(i).cross(h_ref);
```
I think.